.ci/magician/cmd/vcr_merge.go

```diff
@@ -0,0 +1,142 @@
+package cmd
+
+import (
+	"fmt"
+	"magician/exec"
+	"magician/github"
+	"magician/source"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+var vcrMergeCmd = &cobra.Command{
+	Use:   "vcr-merge",
+	Short: "Merge VCR cassettes",
+	Long: `This command is triggered in .ci/gcb-push-downstream.yml to merge vcr cassettes.
+
+	The command expects the following as arguments:
+	1. Reference commit SHA
+
+	It then performs the following operations:
+	1. Get the latest closed PR matching the reference commit SHA.
+	2. Run gsutil to list, copy, and remove the vcr cassettes fixtures.
+	`,
+	Args: cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		reference := args[0]
+		fmt.Println("Reference commit SHA: ", reference)
+		githubToken, ok := os.LookupEnv("GITHUB_TOKEN_CLASSIC")
+		if !ok {
+			fmt.Println("Did not provide GITHUB_TOKEN_CLASSIC environment variable")
+			os.Exit(1)
+		}
+		rnr, err := exec.NewRunner()
+		if err != nil {
+			fmt.Println("Error creating Runner: ", err)
+			os.Exit(1)
+		}
+
+		baseBranch := os.Getenv("BASE_BRANCH")
+		if baseBranch == "" {
+			baseBranch = "main"
+		}
+		gh := github.NewClient(githubToken)
+		execVCRMerge(gh, reference, baseBranch, rnr)
+	},
+}
+
+func execVCRMerge(gh GithubClient, sha string, baseBranch string, runner source.Runner) {
+	arr, err := gh.GetPullRequests("closed", baseBranch, "updated", "desc")
+	if err != nil {
+		fmt.Println("Error getting pull requests: ", err)
+		os.Exit(1)
+	}
+	pr := findPRBySHA(sha, arr)
+	if pr == nil {
+		fmt.Printf("Not finding any matching PR with commit SHA %s\n", sha)
+		return
+	}
+
+	mergeCassettes(false, baseBranch, pr.Number, runner)
+	mergeCassettes(true, baseBranch, pr.Number, runner)
+}
+
+func mergeCassettes(isBeta bool, baseBranch string, prNumber int, runner source.Runner) {
```

Comment:
There are a lot of variables being created that are very similar to each other and only get used once. What would you think about building full paths sooner and then building single-use paths as part of the arguments to a function? For example,

```go
prPath = fmt.Sprintf("%s/refs/heads/auto-pr-%d", prefix, prNumber)
//...
err := listCassettes(prPath + "/fixtures", runner)
//...
cpCassettes(prPath + "/fixtures/*", branchPath + "/fixtures", runner)
//...
rmCassettes(prPath, runner)
```

Reply:
Done